hardware/RegDevice.cpp

```diff
@@ -0,0 +1,23 @@
+#include "stdafx.h"
```

Comment:
Could you make this a (private) struct of the Regulator class ? Saves two extra files, and this could also been a 'RegisterDevice' class looking at the name, i would name it RegulatorDevice